tests/ut/azure_iot_hub_client_properties_ut.c

```diff
@@ -472,110 +473,92 @@ static void testAzureIoTHubClientProperties_GetNextComponentProperty_Success( vo
-    /*Advance to property value */
-    assert_int_equal( AzureIoTJSONReader_NextToken( &xJSONReader ), eAzureIoTSuccess );
-    assert_int_equal( AzureIoTJSONReader_TokenType( &xJSONReader, &xTokenType ), eAzureIoTSuccess );
-    assert_int_equal( xTokenType, eAzureIoTJSONTokenNUMBER );
-    assert_int_equal( AzureIoTJSONReader_GetTokenInt32( &xJSONReader, &lValue ), eAzureIoTSuccess );
-    assert_int_equal( lValue, 42 );
-    assert_int_equal( AzureIoTJSONReader_NextToken( &xJSONReader ), eAzureIoTSuccess );
-
-    /* End of properties */
-    assert_int_equal( AzureIoTHubClientProperties_GetNextComponentProperty( &xTestIoTHubClient,
-                                                                            &xJSONReader,
-                                                                            xResponseType,
-                                                                            xPropertyType,
-                                                                            &ucComponentName,
-                                                                            &ulComponentNameLength ), eAzureIoTErrorEndOfProperties );
+    while( ( xResult = AzureIoTHubClientProperties_GetNextComponentProperty( &xTestIoTHubClient,
+                                                                             &xJSONReader,
+                                                                             xResponseType,
+                                                                             xPropertyType,
+                                                                             &ucComponentName,
+                                                                             &ulComponentNameLength ) ) == eAzureIoTSuccess )
+    {
+        /* Make sure we're on a property name */
+        assert_int_equal( AzureIoTJSONReader_TokenType( &xJSONReader, &xTokenType ), eAzureIoTSuccess );
+        assert_int_equal( xTokenType, eAzureIoTJSONTokenPROPERTY_NAME );
+
+        if( ucComponentName == NULL )
+        {
+            assert_int_equal( ulComponentNameLength, 0 );
+            assert_true( AzureIoTJSONReader_TokenIsTextEqual( &xJSONReader, "not_component", strlen( "not_component" ) ) );
+            /*Advance to property value */
+            assert_int_equal( AzureIoTJSONReader_NextToken( &xJSONReader ), eAzureIoTSuccess );
+            assert_int_equal( AzureIoTJSONReader_TokenType( &xJSONReader, &xTokenType ), eAzureIoTSuccess );
+            assert_int_equal( xTokenType, eAzureIoTJSONTokenNUMBER );
+            assert_int_equal( AzureIoTJSONReader_GetTokenInt32( &xJSONReader, &lValue ), eAzureIoTSuccess );
+            assert_int_equal( lValue, 42 );
+            assert_int_equal( AzureIoTJSONReader_NextToken( &xJSONReader ), eAzureIoTSuccess );
+        }
+        else if( strncmp( "one_component", ucComponentName, strlen( "one_component" ) ) == 0 )
```

Comment:
>ucComponentName [](http://example.com/codeflow?start=43&length=15)

if ucComponentName is one_component"""""", this will still result in valid comparision. Can we also equate lengths?

Reply:
Can we check length in all the places?